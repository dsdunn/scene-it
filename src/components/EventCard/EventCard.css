
.event-card {
  display: flex;
  justify-content: space-between;
  align-items: center;
  padding: 2%;
  text-align: left;
  margin: 2%;
  color:#fffcf9;
  background-color:#261b16;
  opacity: 0.9;
}

.event-card:hover {
  color:#261b16;
  background-color: #f3c225;
}

.selected {
  box-shadow: 0 0 8px 4px #ffa500;
}

.card-label {
  width: 15%;
  display: flex;
  justify-content: center;
}

.circle {
  display:flex;
  justify-content: center;
  align-items: center;
  margin: 2%;
  padding:2%;
  width: 35px;
  height: 35px;
  border-radius: 100%;
  color:#261b16;
  font-size: 1.4em;
  font-weight: bold;
  font-family: 'Shadows Into Light';
  background-color:#fffcf9;
}

.card-title {
  font-size: 1.3em;
  font-weight: 500;
  margin: 3% auto 3% 3%;
  max-height: 50px;
  line-height: 1.2;
  overflow: hidden;
}

.card-venue {
  text-align: center;
  font-size: 1.1em;
  font-weight: 200;
  margin: 1%;
}

.card-info {
  width:80%;
  text-align: left;
}

.date time {
  width: 10%;
  text-align: right;
  font-size: 1.8em;
  font-family: 'Roboto';
  margin: 2%;
}