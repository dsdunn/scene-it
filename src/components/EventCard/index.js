import React from 'react';
import Moment from 'react-moment';

export const EventCard = ({event, selectEvent}) => {
  const handleClick = () => {
    selectEvent(event);
  }
  return (
    <div className="event-card" id={event.id} onClick={handleClick}>
      <h4 className="card-title">{event.title}</h4>
      <h5 className="date">
        <Moment format="ddd MMM Do" date={event.startTime} />
      </h5>
      <p className="card-venue">{event.venueName}</p>
    </div>
    )
}