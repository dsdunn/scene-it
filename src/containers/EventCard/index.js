import React from 'react';

export const EventCard = ({event}) => {
  return (
    <div id={event.id}>
      <h4>{event.title}</h4>
      <p>{event.location}</p>
    </div>
    )
}