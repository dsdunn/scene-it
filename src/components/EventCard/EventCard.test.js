import React from 'react';
import { EventCard } from '../EventCard';
import { shallow } from 'enzyme';

describe('EventCard', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<EventCard event={{id: '3'}}/>)
  })

  it('matches the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  })

})