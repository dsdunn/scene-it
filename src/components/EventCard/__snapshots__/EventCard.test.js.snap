// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EventCard matches the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown
    event={
        Object {
            "id": "3",
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h4 />,
        <p />,
        <p />,
        <p>
          , 
           
</p>,
      ],
      "id": "3",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": "h4",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            undefined,
            ", ",
            undefined,
            " ",
            undefined,
          ],
        },
        "ref": null,
        "rendered": Array [
          undefined,
          ", ",
          undefined,
          " ",
          undefined,
        ],
        "type": "p",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h4 />,
          <p />,
          <p />,
          <p>
            , 
             
</p>,
        ],
        "id": "3",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": "h4",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              undefined,
              ", ",
              undefined,
              " ",
              undefined,
            ],
          },
          "ref": null,
          "rendered": Array [
            undefined,
            ", ",
            undefined,
            " ",
            undefined,
          ],
          "type": "p",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
